dan_layer/engine/src/runtime/impl.rs

```diff
@@ -529,7 +540,7 @@ impl RuntimeInterface for RuntimeInterfaceImpl {
             },
             WorkspaceAction::Take => {
                 let key: Vec<u8> = args.get(0)?;
-                let value = self.tracker.take_from_workspace(&key)?;
+                let value = self.tracker.get_from_workspace(&key)?;
```

Comment:
This naming could be confusing in future

Reply:
Perhaps the WorkspaceAction should be renamed to `Get` ?

Reply:
Oh yeah, good catch, I just ran into a problem with the way we do workspaces and changed that without changing the action - will update in PR #397